### ISSUE TITLE
Cannot parse deps.json if the executable name contains surrogate pair

### ISSUE THREAD

**Issue by huoyaoyuan** [dotnet/runtime#473]
```
PS D:\Code> mkdir 🐎
PS D:\Code> cd 🐎
PS D:\Code\🐎> dotnet new console
The template "Console Application" was created successfully.

Processing post-creation actions...
Running 'dotnet restore' on D:\Code\🐎\🐎.csproj...
  D:\Code\🐎\🐎.csproj 的还原在 97.41 ms 内完成。

Restore succeeded.

PS D:\Code\🐎> dotnet build
用于 .NET Core 的 Microsoft (R) 生成引擎版本 16.4.0-preview-19529-02+0c507a29b
版权所有(C) Microsoft Corporation。保留所有权利。

  D:\Code\🐎\🐎.csproj 的还原在 19.03 ms 内完成。
  你正在使用 .NET Core 的预览版。请查看 https://aka.ms/dotnet-core-preview
  🐎 -> D:\Code\🐎\bin\Debug\netcoreapp3.1\🐎.dll

已成功生成。
    0 个警告
    0 个错误

已用时间 00:00:03.15

PS D:\Code\🐎> dotnet run
A JSON parsing exception occurred in [D:\Code\
Error initializing the dependency resolver: An error occurred while parsing: D:\Code\

PS D:\Code\🐎> .\bin\Debug\netcoreapp3.1\🐎.exe
A JSON parsing exception occurred in [D:\Code\
Error initializing the dependency resolver: An error occurred while parsing: D:\Code\
```

However, it successes if the surrogate pair, or emoji, is in a parent folder name.
```
PS D:\Code\🐂\test> dotnet run
Hello World!
```

**Comment by vitek-karas** [comment]
@lpereira can you please take a look - the JSON parser seems to fail parsing: 
``` JSON
  "targets": {
    ".NETCoreApp,Version=v3.1": {
      "\uD83D\uDC0E/1.0.0": {
        "runtime": {
          "\uD83D\uDC0E.dll": {}
        }
      }
    }
  }
```

There's a secondary issues that it fails to correctly report the error to `stderr`. The error in `stderr` is something like:
```
A JSON parsing exception occurred in [F:\AppModel\repro\
```

While the same error printed to tracing file is:
```
A JSON parsing exception occurred in [F:\AppModel\repro\??\bin\Debug\netcoreapp3.1\??.deps.json]: UTF-16 string is missing low surrogate
```

**Comment by lpereira** [comment]
That's strange; RapidJSON should support surrogate pairs.  Will take a look.

**Comment by mateoatr** [comment]
I have no clue what is causing this, but I cannot repro if I build the installer and use the generated libhostpolicy instead of the 3.1.0 one:

![image](https://user-images.githubusercontent.com/3836488/72548074-d02f9e00-3842-11ea-99cd-8f2de0fe9e5e.png)

**Comment by huoyaoyuan** [comment]
This may be OS-specific. I'm on Windows.
Or may I test against 5.0 nightly?

**Comment by vitek-karas** [comment]
I tried on Windows. It reproes on 3.1 but doesn't repro on 5. Since we've changed the JSON parser for 5 I assume the new one fixes this.

That said the error reporting is still broken for this - when I modify the `.deps.json` to have an error in it (I added `\uu` into one of the values), it will fail to parse (correctly) and the reported error looks like this on the command line:
```
A JSON parsing exception occurred in [F:\AppModel\repro\
Error initializing the dependency resolver: An error occurred while parsing: F:\AppModel\repro\
```

In the trace file the error is better:
```
A JSON parsing exception occurred in [F:\AppModel\repro\??\bin\Debug\netcoreapp3.1\??.deps.json], offset 190 (line 9, column 26): Incorrect hex digit after \u escape in string.
```

Not sure why this is - as apparently dotnet CLI can show this path in console correctly. Maybe the way we write to stderr is not doing the right stuff with UTF-16?
